### ISSUE TITLE
JSON tree

### ISSUE THREAD

**Issue by rossant** [spacetelescope/asdf-standard#100]
Is it conceivable to use JSON instead of YAML in ASDF files? YAML being a "natural superset of JSON", what information would be lost by switching from YAML to JSON in a given ADSF file?

**Comment by mdboom** [comment]
The `tags`, which drive the conversion of parts of the tree to specific datatypes (most importantly ndarrays), and schema validation (by matching a subtree to a particular schema.  This feature, more than the syntax, was the major driving force for using YAML over JSON.

Less importantly, JSON has no comments.

There's some more detail in Section 4 of our recent paper: http://dx.doi.org/10.1016/j.ascom.2015.06.004

That said, one could develop a convention in JSON to losslessly store everything in a YAML file, but it would be cumbersome, e.g.:

```
--- !ndarray
    source: 0
    shape: [256, 256]
    datatype: float64
```

```
{
 '$tag': 'ndarray',
 '$data': {
    'source': 0,
    'shape': [256, 256],
    'datatype': 'float64'
  }
}
```

**Comment by mdboom** [comment]
Also forgot to mention anchors and aliases.  Though those could be replicated with the `JSON Pointer` convention, or just resolved away with duplication as long as the references aren't cyclical.

**Comment by embray** [comment]
Yeah, I don't think a pure JSON tree is going to happen at this stage unless a clear performance benefit can be demonstrated.
